containers.go

```diff
@@ -302,17 +307,19 @@ func (r *containerStore) Load() error {
 	r.bylayer = layers
 	r.byname = names
 	if needSave {
+		if !lockedForWriting {
+			// Eventually, the callers should be modified to retry with a write lock, instead.
+			return errors.New("container store is inconsistent and the current caller does not hold a write lock")
```

Comment:
How can callers know when they need to try again? I mean how to distinguish this error (which requires retry) from any other? I see ErrDuplicateImageNames in layerStore.Load()

Reply:
They can’t, currently.

I plan to add that (“try again”, instead of callers hard-coding specific error values) both to `load` and the callers simultaneously.

Reply:
The retries will be added in #1407 .